### ISSUE TITLE
URLPattern is not picked up in at least some places

### ISSUE THREAD

**Issue by domenic** [w3c/webref#1074]
E.g. https://w3c.github.io/webref/ed/idlnames.json does not have it. However, https://github.com/search?q=repo%3Aw3c%2Fwebref%20urlpattern&type=code shows many matches, so I'm not sure what's going on.

This prevents it from being referenced in Bikeshed specs. https://github.com/speced/bikeshed/issues/2707 https://github.com/WICG/nav-speculation/issues/284 .

**Comment by tabatkins** [comment]
I'm not sure what the reason for this is internally, but yeah, you can see the spec entry in <https://github.com/w3c/webref/blob/bcec95c0ad21a056da64fa408de43a7cfd6ca06b/ed/index.json#L7700>, and it is missing the "dfns"/etc keys that should point to the path of the dfns file (which definitely exists, at <https://github.com/w3c/webref/blob/bcec95c0ad21a056da64fa408de43a7cfd6ca06b/ed/dfns/urlpattern.json>).

If I should be ignoring the presence or absence of these keys and just guessing their URLs, I can, but I was under the impression specs would have those entries iff there was a corresponding file.

**Comment by tidoust** [comment]
Sorry, that's an unfortunate consequence of a bug in the version of the crawler that I released yesterday, which made it fail on all specs in practice.

When a spec cannot be crawled, previous data gets used. Unfortunately, that version of Reffy was also the first one that shipped with the WHATWG URL for the URLPattern spec, so the code failed to set the relevant properties in `ed/index.json` for URLPattern as a result.

Bug in Reffy was fixed in https://github.com/w3c/reffy/pull/1414. Latest `ed/index.json` [now has the right properties](https://github.com/w3c/webref/blob/9e34520de1c52506331e8862c0e9a4588da0356b/ed/index.json#L7681-L7686).

> E.g. https://w3c.github.io/webref/ed/idlnames.json does not have it. 

It still does not because this file is generated from curated data, and the IDL of a couple of recently added specs needs patching before data curation can resume. That's a more usual problem. I'm on it. That should not prevent cross-referencing though: Bikeshed uses the dfn extracts directly, not the curated data.

> If I should be ignoring the presence or absence of these keys and just guessing their URLs, I can, but I was under the impression specs would have those entries iff there was a corresponding file.

I think it's sane to continue to hook into these properties. Hopefully, we won't run into a similar problem any time soon!

We keep data files around in Webref to avoid accidentally deleting files from the repo when a transient network (or crawler!) error occurs. A job runs periodically to make sure that `ed/index.json` is in sync with data files and suggest to delete files when needed (the job typically created https://github.com/w3c/webref/pull/1073 to suggest that all URLPattern data files be deleted since `ed/index.json` no longer pointed to them).
